File: Competitive Programming 3/Section 1.3.3/11661 - Burger Time?/Python/main.py
def main():
    while True:
        l=int(input())
        if l == 0:
            break
        r,d,minDist=-1,-1,l
        s = input()
        for i,c in enumerate(s):
            if c == 'R' or c == 'Z':
                r = i
            if c == 'D' or c =='Z':
                d = i
            if r!=-1 and d!=-1:
                minDist = min(minDist,abs(r-d))
        print(minDist)
main()

File: Competitive Programming 3/Section 1.3.3/11661 - Burger Time?/C++/main.cpp
#include <iostream>

using namespace std;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int l;
    string line;
    while(cin>>l,l){
        int r=-1,d=-1,minDist=l;
        cin >> line;
        for(int i = 0 ; i < line.size();i++){
            if(line[i]=='R' || line[i]=='Z'){
                r=i;
            }
            if(line[i]=='D' || line[i]=='Z'){
                d=i;
            }
            if(r!=-1 && d!=-1){
                minDist = min(minDist,abs(r-d));
            }
        }
        cout << minDist << "\n";
    }
    return 0;
}